src/controllers/user.js

```diff
@@ -32,15 +32,19 @@ export const create = async (req, res) => {
 
 export const getById = async (req, res) => {
   const id = parseInt(req.params.id)
+  const { role } = req.user
 
   try {
-    const foundUser = await User.findById(id)
-
+    let foundUser = {}
+    if (role === 'TEACHER') {
+      foundUser = await findByUniqueAsATeacher(id)
+    } else {
+      foundUser = await User.findById(id, role)
+    }
     if (!foundUser) {
       return sendDataResponse(res, 404, { error: ERR.USER_NOT_FOUND })
     }
-
-    return sendDataResponse(res, 200, foundUser)
+    return res.status(200).json({ status: 'success', data: foundUser })
   } catch (e) {
     return sendMessageResponse(res, 500, 'Unable to get user')
```

Comment:
update the body of this error message so that it follows the same structure as the others, see line 45